examples/jsm/controls/TrackballControls.js

```diff
@@ -538,22 +538,7 @@ class TrackballControls extends EventDispatcher {
 
 			if ( _state === STATE.NONE ) {
 
-				switch ( event.button ) {
-
-					case scope.mouseButtons.LEFT:
-						_state = STATE.ROTATE;
-						break;
-
-					case scope.mouseButtons.MIDDLE:
-						_state = STATE.ZOOM;
-						break;
-
-					case scope.mouseButtons.RIGHT:
-						_state = STATE.PAN;
-						break;
-
-				}
-
+				_state = scope.mouseButtons[ event.button ] || STATE.NONE;
```

Comment:
This code here will always get _state = STATE.NONE.
because scope.mouseButtons[ event.button ] will always equal `undefined`.

```ts
const mouseButtons = { LEFT: STATE.ROTATE, MIDDLE: STATE.DOLLY, RIGHT: STATE.PAN };
console.log(mouseButtons[0]) // result is undefined
```

Reply:
event.button will return the number when the user use mouse buttons to click:

- left button clicked, event.button return 0
- middle wheel clicked, event.button return 1
- right button clicked, event.button return 2

Reply:
Indeed. Fixed that.